exercises/concept/pizza-order/.docs/instructions.md

```diff
@@ -1 +1,46 @@
 # Instructions
+
+You run a pizza shop, and offer three types of pizzas:
+
+- Margherita: $7
+- Caprese: $9
+- Formaggio: $10
+
+If customers want, they can add an unlimited number of extra options: either "ExtraSauce" for $1 or "ExtraToppings" for $2.
+
+Your task is to write code that assists the customer in figuring out the cost to them.
+
+## Calculate the price of a pizza
+
+Provided the pizza's name as the first argument, and an unlimited number of added options, calculate the price of the pizza in dollars.
+
+```js
+pizzaPrice('Margherita');
+// => 7
+
+pizzaPrice('Caprese', 'ExtraSauce', 'ExtraToppings');
+// => 12
+
+pizzaPrice(
+  'Caprese',
+  'ExtraToppings',
+  'ExtraToppings',
+  'ExtraToppings',
+  'ExtraToppings'
+);
+// => 17
+```
+
+## Calculate the total price of an order
+
+Your function will take in a list of `PizzaOrder`s and should return the total price of the order in dollars.
+Each `PizzaOrder` has a `pizza` property - the pizza's name, and an `extras` property - the list of extra options.
+
+```js
+const margherita = new PizzaOrder('Margherita');
+const caprese = new PizzaOrder('Caprese', 'ExtraToppings');
+orderPrice([margherita, caprese]);
+// => 18
+```
+
+You'll realize that you can't write this using recursion, as one test with a tremendous amount of orders will raise a `Maximum call stack size exceeded`. No worries, this is intentional - try implementing this function using an imperative loop!
```

Comment:
I don't understand this:

1. The exemplar solution uses `recursion`.
2. The point of this exercise is to use recursion.
3. Why is this suggesting to use an imperative loop?

Reply:
So this was my confusion about this PR, as I've said in Issue #945. You say: 
> It must be solved using some imperative loop (for loop, .foreach, or best: .reduce)

I first asked a question on the issue, and then after I reread the issue, it seemed that you wanted to teach the student about these errors. I thus proceeded likewise.

> The exemplar solution uses recursion.

No, it uses `.reduce`.
```js
export function orderPrice(pizzaOrders) {
  return pizzaOrders.reduce(
    (result, order) => result + pizzaPrice(order.pizza, ...order.extras),
    0
  );
}
```

> The point of this exercise is to use recursion.

Exactly, which is why I find this to be somewhat convoluted. Should we remove the large test case, maybe? If we do that, it then doesn't teach the errors themselves (about which I agree that it's important to teach).

Reply:
Oh right, yes I was looking at the wrong thing.

Eh. Ok, I'll change the wording a bit. Your intention (and understanding) was correct. We really do want people to hit this "issue" because it is _very_ common when trying to use recursion where you cannot, like in this case.

Thanks!